frontend/pages/SoftwarePage/SoftwareTitleDetailsPage/EditIconModal/EditIconModal.tsx

```diff
@@ -0,0 +1,364 @@
+          className={`${baseClass}__mask-overlay ${baseClass}__mask-overlay--fleet`}
+        />
+      </Card>
+    );
+  };
+
+  const renderPreviewSelfServiceCard = () => {
+    return (
+      <Card
+        borderRadiusSize="medium"
+        color="grey"
+        className={`${baseClass}__preview-card`}
+        paddingSize="xlarge"
+      >
+        <Card
+          className={`${baseClass}__preview-card__self-service`}
+          borderRadiusSize="xxlarge"
+        >
+          <CardHeader
+            header="Self-service"
+            subheader={SELF_SERVICE_SUBHEADER}
+          />
+          <div>
+            <img
+              className={`${baseClass}__preview-img`}
+              src={PreviewSelfServiceIcon}
+              alt="Preview icon on Fleet Desktop > Self-service"
+            />
+          </div>
+          <div className={`${baseClass}__self-service-preview`}>
+            {iconPreviewUrl ? (
+              <img
+                src={iconPreviewUrl}
+                alt="Uploaded self-service icon"
+                style={{
+                  width: 20,
+                  height: 20,
+                  borderRadius: "4px",
+                  overflow: "hidden",
+                }}
+              />
+            ) : (
+              <SoftwareIcon name={software.name} source="apps" size="xsmall" />
+            )}
+            <TooltipTruncatedText value={previewInfo.name} />
+          </div>
+        </Card>
+        <div
+          className={`${baseClass}__mask-overlay ${baseClass}__mask-overlay--self-service`}
+        />
+      </Card>
+    );
+  };
+
+  const renderForm = () => {
+    return (
+      <>
+        <FileUploader
+          canEdit
+          onDeleteFile={onDeleteFile}
+          graphicName="file-png"
+          accept={ACCEPTED_EXTENSIONS}
+          message={UPLOAD_MESSAGE}
+          onFileUpload={onFileSelect}
+          buttonMessage="Choose file"
+          buttonType="link"
+          className={`${baseClass}__file-uploader`}
+          fileDetails={
+            formData && formData.icon
+              ? {
+                  name: formData.icon.name,
+                  description: `Software icon • ${iconDimensions}x${iconDimensions}px`,
+                }
+              : undefined
+          }
+          gitopsCompatible={false}
+        />
+        <h2>Preview</h2>
+        <TabNav>
+          <Tabs selectedIndex={navTabIndex} onSelect={onTabChange}>
+            <TabList>
+              <Tab>
+                <TabText>Fleet</TabText>
+              </Tab>
+              <Tab>
+                <TabText>Self-service</TabText>
+              </Tab>
+            </TabList>
+            <TabPanel>{renderPreviewFleetCard()}</TabPanel>
+            <TabPanel>{renderPreviewSelfServiceCard()}</TabPanel>
+          </Tabs>
+        </TabNav>
+      </>
+    );
+  };
+
+  return (
+    <>
+      <Modal
+        className={baseClass}
+        title={isSoftwarePackage ? "Edit package" : "Edit app"}
```

Comment:
This also needs to be revised on the software edit modal independent of icon edits, right?